### ISSUE TITLE
sql: there should be log/trace events for pre-query tasks

### ISSUE THREAD

**Issue by jordanlewis** [cockroachdb/cockroach#32093]
A recently reported customer problem involved very slow queries due to cold descriptor caches. The traces outputted from these queries were hard to understand, because all of the pre-query activity was "anonymous" in the sense that there were no special log messages explaining what was going on. Instead, the trace forces the reader to guess at the various actions being taken based on the keys being read and written to from the database.

This is fine, because it is possible to infer what's going on based on those keys, but it's pretty annoying when casually reading traces. It would be much better if each "special table" access (read database descriptor, read lease, read table descriptor, etc etc) had an associated log message right before (or even better, a span surrounding it).

I don't think this should be too hard for somebody who knows what's going on in that codebase. Here's my very dumb first attempt (and confirmed it does make the traces easier to understand):

```
diff --git a/pkg/sql/physical_schema_accessors.go b/pkg/sql/physical_schema_accessors.go
index f1e8af210e..004164e884 100644
--- a/pkg/sql/physical_schema_accessors.go
+++ b/pkg/sql/physical_schema_accessors.go
@@ -17,6 +17,8 @@ package sql
 import (
        "bytes"

+       "github.com/cockroachdb/cockroach/pkg/util/log"
+
        "github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
        "github.com/cockroachdb/cockroach/pkg/sql/sqlbase"
        "github.com/cockroachdb/cockroach/pkg/util/encoding"
@@ -49,6 +51,7 @@ func (a UncachedPhysicalAccessor) GetDatabaseDesc(
        name string, flags DatabaseLookupFlags,
 ) (desc *DatabaseDescriptor, err error) {
        desc = &sqlbase.DatabaseDescriptor{}
+       log.Eventf(flags.ctx, "fetching database descriptor %s", name)
        found, err := getDescriptor(flags.ctx, flags.txn, databaseKey{name}, desc)
        if err != nil {
                return nil, err
@@ -126,6 +129,7 @@ func (a UncachedPhysicalAccessor) GetObjectDesc(

        // Look up the table using the discovered database descriptor.
        desc := &sqlbase.TableDescriptor{}
+       log.Eventf(flags.ctx, "fetching object descriptor %s", name.Table())
        found, err := getDescriptor(flags.ctx, flags.txn,
                tableKey{parentID: dbDesc.ID, name: name.Table()}, desc)
        if err != nil {
```

@vivekmenezes could you take a look or hand off? I do think quick action on this will reap benefits.

**Comment by knz** [comment]
taking this, I've been wanting this too

**Comment by knz** [comment]
Fixed by #32257.
